### PR TITLE
WT-5476 Don't write keys when tombstones are globally visible

### PR DESCRIPTION
**Pull request by kommiharibabu**
Whenever a TOMBSTONE update is globally visible, there is no need to reconcile its time pair as stop time pair to the previous update or on-disk value, just discard the key. Along with this whenever a page gets reconciled, check for the obsolete keys and remove them.

### CODE REVIEW

File: src/reconcile/rec_row.c [hunk 1]
```diff
@@ -785,6 +785,13 @@ __wt_rec_row_leaf(
         /* Build value cell. */
         dictionary = false;
         if (upd == NULL) {
+            /* Check the visibility of the on disk key before writing. */
```
Comment (tetsuo-cpp) [inline src/reconcile/rec_row.c]: If we reconcile an update with a globally visible stop time pair AND there are no new updates for that key? Is this for the case where you don't touch a key but its previously written stop pair becomes globally visible while it's in cache?

Either way, could you please elaborate a bit more on the comment?

File: test/suite/test_insert_remove01.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+from helper import copy_wiredtiger_home
+import unittest, wiredtiger, wttest
+from wtdataset import SimpleDataSet
+from wiredtiger import stat
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_insert_remove01.py
+class test_insert_remove01(wttest.WiredTigerTestCase):
```
Comment (tetsuo-cpp) [inline test/suite/test_insert_remove01.py]: Curious about what this tests. Was it just to help with manual testing (so you can run `wt` on it)? Seems like it'd be hard to test that a key got removed from the data file.

Comment (kommiharibabu) [reply test/suite/test_insert_remove01.py]: The test is to hit the code that is written to take care of the globally visible tombstone processing is happening or not? Yes, I agree that currently, we don't have any stats or something to validate the same when it happened.

File: src/reconcile/rec_row.c (hunk 1 above)
Comment (kommiharibabu) [reply src/reconcile/rec_row.c]: Comment has updated to provide a better explanation.

**Pull request merged**